runs/saga_comparison_tests/saga_comparison_tuning.py

```diff
@@ -0,0 +1,214 @@
+import numpy as np
+import click
+import mlflow
+from optuna import Trial
+
+from sklearn.linear_model import Ridge
+from sklearn.utils import Bunch, shuffle
+from sklearn.model_selection import ShuffleSplit
+
+from experiments import Experiment
+from experiments.evaluation import CrossValidate
+from experiments.mlflow import log_experiment
+from experiments.parameter_search import param_space
+from experiments.parameter_search.optuna import OptunaTuner
+from problems import scale_X_y
+
+from suprb import rule, SupRB
+from suprb.logging.combination import CombinedLogger
+from suprb.logging.default import DefaultLogger
+from suprb.logging.stdout import StdoutLogger
+from suprb.optimizer.solution import ga, saga1, saga2, saga3, sas
+from suprb.optimizer.rule import es
+from suprb.solution.initialization import RandomInit
+import suprb.solution.mixing_model as mixing_model
+import suprb
+
+
+random_state = 42
+
+
+def load_dataset(name: str, **kwargs) -> tuple[np.ndarray, np.ndarray]:
+    method_name = f"load_{name}"
+    from problems import datasets
+    if hasattr(datasets, method_name):
+        return getattr(datasets, method_name)(**kwargs)
+
+
+def validate_solution_composition(name: str):
+    valid_names = ['ga', 'saga1', 'saga2', 'saga3', 'sas']
+    if not name in valid_names:
+        raise ValueError('Invalid solution composition')
+
+
+@click.command()
+@click.option('-p', '--problem', type=click.STRING, default='airfoil_self_noise')
+@click.option('-s', '--solution_composition', type=click.STRING, default='ga')
+@click.option('-j', '--job_id', type=click.STRING, default='NA')
+def run(problem: str, solution_composition: str, job_id: str):
+    print(f"Problem is {problem}, with solution composition {solution_composition} and job id {job_id}")
+
+    validate_solution_composition(solution_composition)
+    X, y = load_dataset(name=problem, return_X_y=True)
+    X, y = scale_X_y(X, y)
+    if problem == 'protein_structure':
+        X, y = shuffle(X, y, random_state=random_state, n_samples=9146)
+    else:
+        X, y = shuffle(X, y, random_state=random_state)
+
+    estimator = SupRB(
+        rule_generation=es.ES1xLambda(),
+        n_iter=32, n_rules=4, verbose=10,
+        logger=CombinedLogger([('stdout', StdoutLogger()),
+                               ('default', DefaultLogger())]),)
+
+    tuning_params = dict(
+        estimator=estimator,
+        random_state=random_state,
+        cv=4,
+        n_jobs_cv=4,
+        n_jobs=4,
+        n_calls=1000,
+        timeout=60*60*72,  # 72 hours
+        scoring='neg_mean_squared_error',
+        verbose=10
+    )
+    
+
+    @param_space()
+    def suprb_space(trial: Trial, params: Bunch):
```

Comment:
Move parameters that are the same out of the `if/elif`-structure. That way it becomes better readable and the file doesn't become so big